src/sessions/PageManager.ts

```diff
@@ -108,6 +108,13 @@ export class PageManager {
         } else if (this.page.pageId !== pageId) {
             this.createNextPage(this.page, pageId);
         } else {
+            // If its a resumed or reloaded page, update attributes state in PageManager for event metadata
+            if (this.resumed) {
+                this.collectAttributes(
+                    this.page as Page,
+                    typeof payload === 'object' ? payload : undefined
+                );
+            }
```

Comment:
```suggestion
        } else if (this.resumed) {
            // Update attributes state in PageManager for event metadata
            this.collectAttributes(
                this.page as Page,
                typeof payload === 'object' ? payload : undefined
            );
            return;
        } else {
```

Reply:
Approved, but I agree with this change ;)